pandas/core/indexes/base.py

```diff
@@ -1319,11 +1320,11 @@ def set_names(self, names, level=None, inplace: bool = False):
 
         Parameters
         ----------
-        names : label or list of label
+        names : label or list of label or dict-like for MultiIndex
             Name(s) to set.
         level : int, label or list of int or label, optional
-            If the index is a MultiIndex, level(s) to set (None for all
-            levels). Otherwise level must be None.
+            If the index is a MultiIndex and names is not dict-like, level(s) to set
```

Comment:
pls add version changed to both of these

Reply:
I've added 1.2, if this is not ready I will change to 1.3